chain/ethereum/ethereum.go

```diff
@@ -183,6 +186,66 @@ func ChunkedFilterLogs(ctx context.Context, client ETHClient, addresses []common
 	return failedQueries, nil
 }
 
+// BatchedFilterLogs queries the blockchain for past events in batches.
+// It slices addresses and total number of blocks with pre-defined batch size.
+// The results are later fed into a log chan that was provided by the caller.
+// Failed query intervals are retried after a backoff period, increased exponentially.
+func BatchedFilterLogs(ctx context.Context, client ETHClient, addresses []common.Address, fromBlock, toBlock uint64, logChan chan<- types.Log, backoff time.Duration) {
+	if backoff == 0 || backoff > maxBackoffDuration {
+		backoff = initialBackoffDuration
```

Comment:
Why do you reset `backoff` to `initialBackoffDuration` when `backoff > maxBackoffDuration`? I think `backoff` should be set to `maxBackoffDuration` if `backoff > maxBackoffDuration`.